api/v1alpha1/fluxdeployer_types.go

```diff
@@ -14,9 +15,21 @@ type FluxDeployerSpec struct {
 	// +required
 	SourceRef ObjectReference `json:"sourceRef"`
 
+	// The interval at which to reconcile the Kustomization.
+	// +kubebuilder:validation:Type=string
+	// +kubebuilder:validation:Pattern="^([0-9]+(\\.[0-9]+)?(ms|s|m|h))+$"
+	// +required
+	Interval metav1.Duration `json:"interval"`
+
 	// +kubebuilder:pruning:PreserveUnknownFields
 	// +kubebuilder:validation:Schemaless
-	KustomizationTemplate kustomizev1.KustomizationSpec `json:"kustomizationTemplate"`
+	// +optional
+	KustomizationTemplate *kustomizev1.KustomizationSpec `json:"kustomizationTemplate,omitempty"`
```

Comment:
TODO: Add check that either one _MUST_ be defined.

Reply:
we should define which one take precedence.